File: app/src/main/java/com/ducktapedapps/updoot/utils/accountManagement/AuthenticationService.java
package com.ducktapedapps.updoot.utils.accountManagement;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

public class AuthenticationService extends Service {
    private Authenticator mAuthenticator;

    @Override
    public void onCreate() {
        // Create a new authenticator object
        mAuthenticator = new Authenticator(this);
    }

    @Override
    public IBinder onBind(Intent intent) {
        return mAuthenticator.getIBinder();
    }
}

File: app/src/main/java/com/ducktapedapps/updoot/utils/accountManagement/AuthenticationService.kt
package com.ducktapedapps.updoot.utils.accountManagement

import android.app.Service
import android.content.Intent
import android.os.IBinder

class AuthenticationService : Service() {
    private lateinit var mAuthenticator: Authenticator
    override fun onCreate() {
        mAuthenticator = Authenticator(this)
    }

    override fun onBind(intent: Intent): IBinder? {
        return mAuthenticator.iBinder
    }
}